docs/core.rst

```diff
@@ -0,0 +1,86 @@
+Core
+===================
+The DSI Core middleware defines the Terminal concept. An instantiated Terminal is the human/machine DSI interface. The person setting up a Core Terminal only needs to know how they want to ask questions, and what metadata they want to ask questions about. If they don’t see an option to ask questions the way they like, or they don’t see the metadata they want to ask questions about, then they should ask a Driver Contributor or a Plugin Contributor, respectively.
+
+A Core Terminal is a home for Plugins, and an interface for Drivers. A Core Terminal is instantiated with a set of default Plugins and Drivers, but they must be loaded before a user query is attempted. Here's an example of how you might work with DSI using an interactive Python interpreter for your data science workflows::
+
+>>> from dsi.core import Terminal
+>>> a=Terminal()
+>>> a.list_available_modules('plugin')
+>>> # ['Bueno', 'Hostname', 'SystemKernel']
+>>> a.load_module('plugin','Bueno','consumer',filename='./data/bueno.data')
+>>> # Bueno plugin consumer loaded successfully.
+>>> a.load_module('plugin','Hostname','producer')
+>>> # Hostname plugin producer loaded successfully.
+>>> a.list_loaded_modules()
+>>> # {'producer': [<dsi.plugins.env.Hostname object at 0x7f21232474d0>], 
```

Comment:
I think I like the full path to the plugin Class because it tells us that the producer functionality is implemented as a Plugin. To a Core Terminal user, this could be a good reminder that the thing they loaded is a plugin vs a driver. It also gives us the ability to throw errors if some non-sense is requested, like a plugin for a front-end.